method.rb

```diff
@@ -0,0 +1,218 @@
+        status_cumulator &&= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    else
+      loop do
+        status_cumulator &&= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    end
+    status_cumulator
+  end
+
+  def my_any?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/AbcSize
+    status_cumulator = false
+    i = 0
+    bg = block_given?
+    if bg == true && parameter.nil?
+      loop do
+        status_cumulator ||= yield self[i]
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter.is_a?(Regexp)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter.is_a?(Class)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter_condition?(parameter)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == false && parameter.nil?
+      loop do
+        break if i == size || status_cumulator == true
+
+        status_cumulator ||= !self[i].nil?
+        i += 1
+      end
+    elsif bg == false && parameter.is_a?(Regexp)
+      loop do
+        status_cumulator ||= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == false && parameter.is_a?(Class)
+      loop do
+        status_cumulator ||= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    else
+      loop do
+        status_cumulator ||= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    end
+    status_cumulator
+  end
+
+  def my_none?(parameter = nil)
+    !my_any?(parameter)
+  end
+
+  def my_count(parameter = nil)
+    counter = 0
+    if parameter
+      my_each { |item| counter += 1 if item == parameter }
+      counter
+    elsif block_given?
+      my_each { |item| counter += 1 if yield item }
+      counter
+    else
+      size
+    end
+  end
+
+  def my_inject(parameter = nil)
+    return to_enum(:my_inject) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    if parameter
+      cumulator = parameter
+      i = 0
+    else
+      cumulator = output_array[0]
+      i = 1
+    end
+    while i < size
+      cumulator = yield cumulator, output_array[i]
+      i += 1
+    end
+    cumulator
+  end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should work with symbols as arguments (:+) and perform the right operation.